crates/nu-parser/src/parser.rs

```diff
@@ -3704,12 +3704,21 @@ pub fn parse_signature_helper(working_set: &mut StateWorkingSet, span: Span) ->
                                         working_set.set_variable_type(var_id.expect("internal error: all custom parameters must have var_ids"), Type::List(Box::new(syntax_shape.to_type())));
                                         *shape = syntax_shape;
                                     }
-                                    Arg::Flag(Flag { arg, var_id, .. }) => {
-                                        // Flags with a boolean type are just present/not-present switches
-                                        if syntax_shape != SyntaxShape::Boolean {
-                                            working_set.set_variable_type(var_id.expect("internal error: all custom parameters must have var_ids"), syntax_shape.to_type());
-                                            *arg = Some(syntax_shape)
+                                    Arg::Flag(Flag {
+                                        arg,
+                                        var_id,
+                                        default_value,
+                                        ..
+                                    }) => {
+                                        if syntax_shape == SyntaxShape::Boolean {
+                                            // set up default value, this would allow something
+                                            // def a [--x: bool] { $x }; a
+                                            *default_value =
+                                                Some(Value::bool(false, Span::unknown()));
                                         }
```

Comment:
Should `bool` still be special-cased here? The original special casing is what led to the odd behavior discussed in #7260 ([my comment](https://github.com/nushell/nushell/issues/7260#issuecomment-1728337485)) and complicated the logic that eventually led to #10410 and then #10450.

This still leads to a one-off special behavior that

```nu
def foo [--x: T] { $x | describe }
foo
```

prints `nothing` in all cases _except_ for `T = bool`, and leaves no way to leave a boolean argument as un-specified by the caller (i.e. `foo` cannot discern between `(foo)` and `(foo --x=false)`).

Reply:
Hi, thanks for your comment!

Yeah I'm also thinking about what's the value of `$x` in `foo` call.  But finally I decide to make it false because it doesn't introduce a larger breaking change.  But yeah, your opinion is valid, currently I'm not really sure about it..

Reply:
After a discussion with core team, I change the value to `null` to make it consistent with other types.